Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs

```diff
@@ -20,10 +21,17 @@ public class GenericDictionaryEquivalencyStep : IEquivalencyStep
         if (comparands.Expectation != null)
         {
             Type expectationType = comparands.GetExpectedType(context.Options);
+
             if (DictionaryInterfaceInfo.FindFrom(expectationType, "expectation") is { } expectedDictionary)
             {
+                if (SubjectIsNonGenericDictionary(comparands.Subject))
+                {
+                    // Because we handle non-generic dictionaries later
+                    return EquivalencyResult.ContinueWithNext;
+                }
+
                 if (AssertSubjectIsNotNull(comparands.Subject)
-                    && EnsureSubjectIsDictionary(comparands, expectedDictionary) is { } actualDictionary)
+                    && EnsureSubjectIsGenericDictionary(comparands, expectedDictionary) is { } actualDictionary)
```

Comment:
🤔 I think renaming this to `EnsureSubjectIsOfTheExpectedDictionaryType` makes it clearer why there is also a `SubjectIsNonGenericDictionary`